cdap-ui/app/directives/search/search-ctrl.js

```diff
@@ -22,11 +22,13 @@ angular.module(PKG.name + '.commons')
     this.onSearch = function(event) {
       if (event.keyCode === 13) {
         $state.go('search.objectswithtags', {tag: this.searchTxt});
+        this.searchTxt = '';
```

Comment:
you have to set this.displaySearchBox to false as well, otherwise the box won't close